messenger_test.go

```diff
@@ -738,6 +738,78 @@ func (s *MessengerSuite) TestContactPersistence() {
 	s.Require().Equal(expectedContact, actualContact)
 }
 
+func (s *MessengerSuite) TestVerifyENSNames() {
+	rpcEndpoint := os.Getenv("RPC_ENDPOINT")
+	if rpcEndpoint == "" {
+		s.T().Skip()
+	}
+	contractAddress := "0x314159265dd8dbb310642f98f50c066173c1259b"
+	pk1 := "04325367620ae20dd878dbb39f69f02c567d789dd21af8a88623dc5b529827c2812571c380a2cd8236a2851b8843d6486481166c39debf60a5d30b9099c66213e4"
+	pk2 := "044580b6aef9ddebd88c373b43c91237dcc95a8307bc5837d11d3ad2fa5d1dc696b598e7ccc498b414ba80b86b129c48e1eb9464cc9ea26224321539f2f54024cc"
+	pk3 := "044fee950d9748606da2f77d3c51bf16134a59bde4903aa68076a45d9eefbb54182a24f0c74b381bad0525a90e78770d11559aa02f77343d172f386e3b521c277a"
+	pk4 := "not a valid pk"
+
+	ensDetails := []EnsDetails{
```

Comment:
It could be rewritten in this way:
```go
testCases := []struct{
    Name sting 
    ENSDetails ENSDetails
    Error err
    ENSResponse ENSResponse
}{
    {
        Name: "valid",
        EnsDetails: EnsDetails{Name: "pedro.stateofus.eth", PublicKey: pk1},
        EnsResponse: ...,
    },
    {
        Name: "pk not matching name",
        EnsDetails{Name: "pedro.stateofus.eth", PublicKey: pk2},
        EnsResponse: ...,
    },
    { ... },
}

for _, tc := range testCases {
    s.Run(t, func() {
        response, err := s.m.VerifyENSNames(tc.ENSDdetails, rpcEndpoint, contractAddress)
        if tc.Error != nil {
            s.EqualError(tc.Error, err.Error())
        }
        s.EqualValues(tc.ENSResponse, response)
    })
}
```